server/tests/test_command_safety_adversarial.py

```
@@ -0,0 +1,252 @@
+    branch (LLM raises because the langchain stub doesn't implement invoke)
+    and falsely report "passed" when the assertion expects conclusion=True.
+    """
+    import sys
+    from unittest.mock import MagicMock
+
+    if isinstance(sys.modules.get("langchain"), MagicMock):
+        pytest.skip("conftest.py stubs langchain; run integration tests in a clean env")
+
+
+@pytest.fixture(autouse=True)
+def _set_required_env(monkeypatch):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_set_required_env`

Add return type annotation: `None`

(ANN202)